api/src/main/java/org/openmrs/module/openconceptlab/importer/Saver.java

```diff
@@ -665,13 +656,29 @@ private void updateOrAddNamesFromOcl(Concept concept, OclConcept oclConcept) {
 					name.setUuid(version5Uuid(oclConcept.getUrl() + "/names/" + oclName.getUuid()).toString());
 				}
 				name.setConceptNameType(oclNameType);
-				name.setLocalePreferred(oclName.isLocalePreferred());
+
+				// If this concept does not yet have any names preferred in the given locale, make it preferred
+				// This is needed in order to match expected state in the Concept.addName method
+				boolean preferred = oclName.isLocalePreferred();
+				if (!preferred) {
+					preferred = !hasExistingPreferredNameInExactLocale(concept, oclName.getLocale());
+				}
+				name.setLocalePreferred(preferred);
 
 				concept.addName(name);
 			}
 		}
 	}
 
+	private boolean hasExistingPreferredNameInExactLocale(Concept concept, Locale locale) {
+		for (ConceptName cn : concept.getNames()) {
+			if (BooleanUtils.isTrue(cn.getLocalePreferred()) && cn.getLocale().equals(locale)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
```

Comment:
The may be a bit repetitive of what is in the core Concept.getNames(Locale) method, but one of the reasons why this bug emerged is that the implementation of these methods in core has changed over time and isn't always clear when the Locale is being matched exactly and when it falls back to Language.  I didn't want to leave this to chance, so just am implementing a version here that contains exactly the logic we need.